aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -279,9 +288,24 @@ public void onResult(SignUpResult result) {
 
                 @Override
                 public void onError(Exception error) {
-                    onException.accept(
-                        new AuthException("Sign up failed", error, "See attached exception for more details")
-                    );
+                    if (error instanceof UsernameExistsException) {
+                        onException.accept(
+                                new AuthException.AWSCognitoAuthException.UsernameExistsException(error.getCause())
+                        );
+                    } else if (error instanceof AliasExistsException) {
+                        onException.accept(
+                                new AuthException.AWSCognitoAuthException.AliasExistsException(error.getCause())
```

Comment:
What's the difference between AliasExistsException and UsernameExistsException? Might be better to check for either AliasExists or UsernameExists and return either as our UsernameExists exception.

Reply:
AliasExistsException is thrown when a user tries to confirm the account with an email or phone number that has already been supplied as an alias from a different account. This exception tells user that an account with this email or phone already exists.

UsernameExistsException is thrown when Amazon Cognito encounters a user name that already exists in the user pool.

So, both of them should be kept.